src/resources/views/crud/buttons/quick.blade.php

```diff
@@ -37,3 +56,56 @@
         {{ $label }}
     </{{ $wrapper['element'] }}>
 @endif
+
+
+@if($buttonAjaxConfiguration)
+{{-- Button Javascript --}}
+{{-- Pushed to the end of the page, after jQuery is loaded --}}
+@push('after_scripts') @if (request()->ajax()) @endpush @endif
+@bassetBlock('backpack/crud/buttons/quick-button.js')
+<script>
+	if (typeof sendQuickButtonAjaxRequest != 'function') {
+	  $("[data-button-type=quick-ajax]").unbind('click');
+
+	  function sendQuickButtonAjaxRequest(button) {
+		var route = $(button).attr('data-route');
+				$.ajax({
+			      url: route,
+			      type: $(button).attr('data-method'),
+			      success: function(result) {
+                    if($(button).attr('data-refresh-table') && typeof crud != 'undefined' && typeof crud.table != 'undefined'){
+                        crud.table.draw(false);
+                    }
+                    let message;
+                    if(result.message){
+                        //if message is returned from the API
+                        message = result.message;
+                    }else{
+                        //if message is not returned from the API
+                        let buttonSuccessTitle = button.getAttribute('data-success-title');
+                        let buttonSuccessMessage =  button.getAttribute('data-success-message');
+                        message = `<strong>${buttonSuccessTitle}</strong><br/>${buttonSuccessMessage}`;
+                    }
```

Comment:
```suggestion
                    let buttonSuccessTitle = button.getAttribute('data-success-title');
                    let buttonSuccessMessage =  button.getAttribute('data-success-message');
                    let message = `<strong>${buttonSuccessTitle}</strong><br/>${buttonSuccessMessage}`
                    if(result.message){
                        // if message is returned from the API use it instead of the default message
                        message = result.message;
                    }
                    
```

Reply:
@pxpm I intentionally went that way to save browser processing, to make it lighter.

If the result came from API, the Browser will not have to execute:
```javascript
let buttonSuccessTitle = button.getAttribute('data-success-title');
let buttonSuccessMessage =  button.getAttribute('data-success-message');
let message = `<strong>${buttonSuccessTitle}</strong><br/>${buttonSuccessMessage}`
```

Reply:
Sure, there is nothing wrong with your approach. 
I wouldn't have worried about saving this work from the browser. The "hard-work" is already done, here we have an html element (button object) and we just get some properties of that object to write a string. It's very very cheap work

If we were sending a request to the server or something similar, I would avoid making that request unless strictly necessary but in this case ... méh. 

But since we are exploring possibilities, how about this then:
```suggestion
                    let message;
                    let defaultMessage = function(button) {
                        // since this is inside a closure, we don't execute until we manually call it.
                        let buttonSuccessTitle = button.getAttribute('data-success-title');
                        let buttonSuccessMessage =  button.getAttribute('data-success-message');
                        return `<strong>${buttonSuccessTitle}</strong><br/>${buttonSuccessMessage}`;
                    }
                    
                    //if message is returned from the API use that message
                    if(result.message){
                        message = result.message;
                    }
                    
                    // if message variable has no value, we build the default message, otherwise use the previous set value without calling defaultMessage()
                    message ??= defaultMessage(button);
```

Reply:
Here, our objective is to make the code look simpler.

Let's see it with the eyes of a beginner. Do you think a closure looks simple? I feel that concatenation looks simpler.

Reply:
I don't get sorry, it's because I called it `closure` and not `function` ? Basically it's what it is, a **function** (a block of code you create and defer to execute at any given time in the future), in this case, an **anonymous function**, where you don't need to define visibility or names. Just store the function in a variable, and invoke it if needed. 

This would be pretty much the same as having: 
```javascript
function defaultMessage(button) {
// ...
}

message ??=defaultMessage(button); 
```

The usage of a closure allow us to don't polute the global space with "specific functions" that only belong to this bit of javascript. 

I am sorry if this doesn't feel good for a beginner, but this is as beginner as it can get.